src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/api/NodesDnApiAction.java

```diff
@@ -0,0 +1,150 @@
+ * By default the APIs are disabled and can be enabled by a YML setting - {@link ConfigConstants#OPENDISTRO_SECURITY_NODES_DN_DYNAMIC_CONFIG_ENABLED}
+ *
+ * The backing data is stored in {@link ConfigConstants#OPENDISTRO_SECURITY_CONFIG_INDEX_NAME} which is populated during bootstrap.
+ * For existing clusters, {@link com.amazon.opendistroforelasticsearch.security.tools.OpenDistroSecurityAdmin} tool can
+ * be used to populate the index.
+ *
+ * See {@link com.amazon.opendistroforelasticsearch.security.dlic.rest.api.NodesDnApiTest} for usage examples.
+ */
+public class NodesDnApiAction extends PatchableResourceApiAction {
+    public static final String STATIC_ES_YML_NODES_DN = "STATIC_ES_YML_NODES_DN";
+    private final List<String> staticNodesDnFromEsYml;
+
+    @Inject
+    public NodesDnApiAction(final Settings settings, final Path configPath, final RestController controller, final Client client,
+        final AdminDNs adminDNs, final ConfigurationRepository cl, final ClusterService cs,
+        final PrincipalExtractor principalExtractor, final PrivilegesEvaluator evaluator, ThreadPool threadPool, AuditLog auditLog) {
+        super(settings, configPath, controller, client, adminDNs, cl, cs, principalExtractor, evaluator, threadPool, auditLog);
+        this.staticNodesDnFromEsYml = settings.getAsList(ConfigConstants.OPENDISTRO_SECURITY_NODES_DN, Collections.emptyList());
+    }
+
+    @Override
+    protected void registerHandlers(RestController controller, Settings settings) {
+        if (settings.getAsBoolean(ConfigConstants.OPENDISTRO_SECURITY_NODES_DN_DYNAMIC_CONFIG_ENABLED, false)) {
+            controller.registerHandler(Method.GET, "/_opendistro/_security/api/nodesdn/{name}", this);
+            controller.registerHandler(Method.GET, "/_opendistro/_security/api/nodesdn/", this);
+            controller.registerHandler(Method.DELETE, "/_opendistro/_security/api/nodesdn/{name}", this);
+            controller.registerHandler(Method.PUT, "/_opendistro/_security/api/nodesdn/{name}", this);
+            controller.registerHandler(Method.PATCH, "/_opendistro/_security/api/nodesdn/", this);
+            controller.registerHandler(Method.PATCH, "/_opendistro/_security/api/nodesdn/{name}", this);
+        }
+    }
+
+    @Override
+    protected void handleApiRequest(RestChannel channel, RestRequest request, Client client) throws IOException {
+        if (!isSuperAdmin()) {
+            forbidden(channel, "API allowed only for admin.");
+            return;
+        }
+        super.handleApiRequest(channel, request, client);
+    }
+
+    protected void consumeParameters(final RestRequest request) {
+        request.param("name");
+        request.param("show_all");
+    }
+
+    @Override
+    protected boolean isReservedAndAccessible(SecurityDynamicConfiguration<?> existingConfiguration, String name) {
+        if (STATIC_ES_YML_NODES_DN.equals(name)) {
+            return false;
+        }
```

Comment:
> couldn't get this condition here.
> Reserved is a type of configuration (roles, users, mapping etc) which can be only updated by super admin.
> why are we checking ES_YML_NODES_DN here?

It is an amalgamation of super-admin and **accessible**. Since this is static coming from YML, the intention is to not allow updates to this key.